compiler/rustc_data_structures/src/sync.rs

```diff
@@ -54,7 +59,43 @@ pub use std::sync::atomic::Ordering::SeqCst;
 pub use vec::AppendOnlyVec;
 
 mod vec;
+use parking_lot::lock_api::RawMutex as _;
+use parking_lot::lock_api::RawRwLock as _;
+use parking_lot::{RawMutex, RawRwLock};
 
+mod mode {
+    use super::Ordering;
+    use std::sync::atomic::AtomicU8;
+
+    const UNINITIALIZED: u8 = 0;
+    const INACTIVE: u8 = 1;
+    const ACTIVE: u8 = 2;
+
+    static MODE: AtomicU8 = AtomicU8::new(UNINITIALIZED);
+
+    #[inline]
+    pub fn active() -> bool {
```

Comment:
The names in this module are extremely generic:
```
mod mode    
static MODE    
const UNINITIALIZED/ACTIVE/INACTIVE 
fn active()
fn set()
```
They tell the reader almost no information. I suggest these more descriptive names:
```
mod threadedness
static THREADEDNESS
const UNINITIALIZED/SINGLE/MULTI
fn is_single_threaded() / is_multi_threaded()
fn set_threadedness()
```

Reply:
This was decided in https://github.com/rust-lang/rust/pull/107586#discussion_r1122591336 and some comments below that.
This is to use `sync::active` to indicate that only the data structures in the `sync` module are involved, not the entire compiler (whether in parallel mode or not).

Reply:
I disagree with that decision. If I see code like this:
```
if rustc_data_structures::sync::active() { ... }
```
it is unclear what `active` means. When I first looked at the code, I didn't understand it, and I had to go find the definition of `active` to understand it.

In comparison, this:
```
if rustc_data_structures::sync::is_multi_threaded() { ... }
```
is much clearer. Good names are important!

Reply:
Yea makes sense. Let me make a change and add a comment in #107586

Reply:
They control thread safetly, not thread count so `is_multi_threaded` is a bit misleading.

Reply:
Maybe `dyn_thread_safe` or `is_dyn_thread_safe`?

Reply:
`Lock` also has a field called `single_thread`. Should it also be called `is_dyn_thread_safe`

Reply:
I agree with `is_dyn_thread_safe`